samza-core/src/main/java/org/apache/samza/clustermanager/ContainerManager.java

```diff
@@ -285,34 +297,41 @@ void handleExpiredRequest(String processorId, String preferredHost,
    * @param containerAllocator to request physical resources
    */
   public void registerContainerPlacementAction(ContainerPlacementRequestMessage requestMessage, ContainerAllocator containerAllocator) {
-    LOG.info("Received a ContainerPlacement action request: {}", requestMessage);
     String processorId = requestMessage.getProcessorId();
     String destinationHost = requestMessage.getDestinationHost();
-    Pair<ContainerPlacementMessage.StatusCode, String> actionStatus =
-        checkValidControlAction(processorId, destinationHost, requestMessage.getUuid());
-
+    Optional<Pair<ContainerPlacementMessage.StatusCode, String>> actionStatus = validatePlacementAction(requestMessage);
+    if (!actionStatus.isPresent()) {
+      // Action is supposed to be queued
+      LOG.info("ContainerPlacement request is en-queued metadata: {}", requestMessage);
+      return;
+    }
+    LOG.info("ContainerPlacement action is de-queued metadata: {}", requestMessage);
+    // Remove the request message from metastore since this message is already acted upon
+    containerPlacementMetadataStore.deleteContainerPlacementRequestMessage(requestMessage.getUuid());
+    // Action is de-queued upon so we record it in the cache
```

Comment:
What if the machine bearing the AM fails at line 311, 
will the action be silently deleted/dropped?